### DISCUSSION TITLE
ESP32 losing configuration

### DISCUSSION THREAD

**Discussion by SimonFili** [arendst/Tasmota#19408] [Support request] [answered]
Hi all,  

I bought a module from Rocket Controller, which is en ESP32 + 4 relay and an expander with 8 more relays through an i2C with the PCF8574 chip.  

So I compiled a version (13.1.0.1) of ESP32 with the PCF8574 and some EZO chip enabled.

Issues: 
When I enable the 4 relays, button and basic thing, it works fine.
If I enable i2C alone, I see the 8 relays and they work fine
If I enable both, the config does not save and after the reboot, I get an empty config page.

Here's some screenshot 
1) the 4 relay config, this works fine:
![image](https://github.com/arendst/Tasmota/assets/7669537/e9e10439-089d-4343-beff-b669d4b6cce7)

2) when I enable those 2 GPIO for i2C, config does not save and I have an empty config page:
![image](https://github.com/arendst/Tasmota/assets/7669537/97750634-b141-4024-af48-75af76edde90)

3) If I only configure the i2c like 2 above, it's working fine, but only for the 8 relays on the extender.

I've put weblog 4 to try to capture something, but I did not see anything usefull.

When I have only the i2c configured, I can see this page:
![image](https://github.com/arendst/Tasmota/assets/7669537/9eec318b-6da8-4149-995e-ca8fd7bec674)

If I enable first i2c, all 8 relays work.  Then I can enable relay 1 (GPIO25), relay 2 (GPIO26).
**If I then enable relay 3 (GPIO 33) I lose the config after the restart.**

Any help would be appreciated! Weird!

**Comment by barbudor** [comment]
What are the i2c addresses of the EZO chips?
Do they conflict with the PCF?
Please provide output of i2cdriver commands

**Reply by barbudor** [reply]
Can you confirm which model of expander you have ?
AFAIK, expanders are 4 relays + 4 inputs, not 8 relays so you should have 

You should also disable the RFRecv on GPIO27 as I believe you don't have this optin. It tends to create a lot of problems and false interrupts.

Please also read this where hte devices has been discussed in many details

**Reply by sfromis** [reply]
Tasmota losing config after a restart is typically due to a boot loop where Tasmota will remove config details till able to boot without crashing.

**Reply by SimonFili** [reply]
I don't have the EZO connected yet.
You may be right about the RF chip, I will try to disable it.

I did a full reset, reload the config with i2c only:
21:51:07.492 RSL: RESULT = {"I2CScan":"Device(s) found on bus1 at 0x21"}
I can access all 8 relays

Then loaded the config with the 4 relays, removed the RFrecv on GPI027
Works fine for the 4 relays

The added the i2c config... same issue, rebooted with no configuration.
Here's what I have:
![image](https://github.com/arendst/Tasmota/assets/7669537/5999aea9-1488-4b41-8778-233297828853)

**Reply by barbudor** [reply]
Ah, I thought it was the Astra family, not the ESP32R4
I have the Astra family

I will try later this week to test the same configuration
Could you please share the details of your user_config_override.h (except any credentials of course) so I can try to reproduce the same firmware ?

**Reply by SimonFili** [reply]
Here's what I documented for myself:

Following [instructions ](https://github.com/arendst/Tasmota/discussions/18745)removed comments on those lines in my_user_config.h 
#define USE_PCF8574 
#define USE_PCF8574_SENSOR
#define USE_PCF8574_MQTTINPUT
#define PCF8574_ADDR1 0x20
#define PCF8574_ADDR1_COUNT 7


and those line for the  EZO:
#define USE_EZOPH
#define USE_EZOORP
#define USE_EZOEC

**Reply by SimonFili** [reply]
@sfromis is there any way to capture this in the logs to see where the issue is?
@barbudor I will work on this again today, did you see my message saying that when I activate a third relay with the i2c enabled, I get into the issue? 

Any ideas you may have for me to try, I would appreciate.

Thanks!

**Comment by SimonFili** [comment]
Here's what I tried just now:

I found this template from the restore of a configuration from the vendor:
{"NAME":"ESP32R4","GPIO":[1,1,576,1,480,1,1,1,1,1,1,1,1,1,1,1,0,640,608,1,0,224,225,1152,0,0,0,0,227,226,32,33,34,0,0,35],"FLAG":0,"BASE":1}

I went to configure other and pasted the template and saved.
I can see this now in the template:
![image](https://github.com/arendst/Tasmota/assets/7669537/df6969cb-6b88-490a-8f7b-907994a44491)

Now I go to configure module and try to apply this ESP32R4 (0) template: 
I get this in the logs:
```
14:03:55.290 HTP: Configure Module
14:04:11.435 WIF: Checking connection...
14:04:31.449 WIF: Checking connection...
14:04:38.075 SRC: WebGui from 192.168.10.110
14:04:38.076 CMD: Grp 0, Cmd 'BACKLOG', Idx 0, Len 11, Pld -99, Data 'Module;GPIO'
14:04:38.146 HTP: Restart
14:04:38.147 SRC: WebGui from 192.168.10.110
14:04:38.153 SRC: Backlog
14:04:38.154 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 0, Pld -99, Data ''
14:04:38.157 RSL: RESULT = {"Module":{"0":"ESP32R4"}}
14:04:38.163 SRC: Backlog
14:04:38.165 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 0, Pld -99, Data ''
14:04:38.170 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO3":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO23":{"0":"None"}}
14:04:38.319 CFG: Saved, Count 73, Bytes 4096
```

Then a restart with the config not applied and this log:
```
0:00:00.003 HDW: ESP32-D0WD (PSRAM disabled)
00:00:00.004 HDW: PSRAM is disabled, requires specific compilation on this hardware (see doc)
00:00:00.076 UFS: FlashFS mounted with 12572 kB free
00:00:00.103 CFG: Loaded from File, Count 73
00:00:00.105 FRC: Some settings have been reset (6)
00:00:00.107 CFG: CR 380/699, Busy 0
00:00:00.122 ROT: Mode 1
00:00:00.128 TFS: File '.drvset003' not found
00:00:00.128 CFG: Energy use defaults as file system not ready or file not found
00:00:00.131 SRC: Restart
00:00:00.133 Project tasmota - Tasmota Version 13.1.0.1(tasmota32)-2_0_11(2023-08-15T15:15:03)
00:00:00.134 ETH: No ETH MDC and/or ETH MDIO GPIO defined
00:00:00.322 CFG: Saved, Count 74, Bytes 4096
00:00:00.423 WIF: Checking connection...
00:00:00.424 WIF: Attempting connection...
00:00:00.878 WIF: Connecting to AP1 Simon Channel 6 BSSId 98:DA:C4:A4:46:F2 in mode 11n as tasmota-AE1A64-6756...
00:00:00.874 WIF: Checking connection...
00:00:00.875 WIF: Attempting connection...
00:00:02.716 WIF: IPv4 192.168.10.128, mask 255.255.255.0, gateway 192.168.10.1
00:00:02.780 WIF: Checking connection...
00:00:02.781 WIF: Connected
00:00:02.881 NTP: Sync time...
00:00:02.917 WIF: DNS resolved '2.pool.ntp.org' (209.115.181.107) in 35 ms
00:00:02.983 RTC: UTC 2023-08-30T13:04:47, DST 2023-03-26T02:00:00, STD 2023-10-29T03:00:00
14:04:47.000 RTC: Synced by NTP
14:04:47.044 HTP: Web server active on tasmota-AE1A64-6756 with IP address 192.168.10.128
14:04:48.049 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"13.1.0.1(tasmota32)","FallbackTopic":"cmnd/DVES_AE1A64_fb/","GroupTopic":"cmnd/tasmotas/"}}
14:04:48.062 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-AE1A64-6756","IPAddress":"192.168.10.128","IP6Global":"","IP6Local":"fe80::3ee9:eff:feae:1a64"}}
14:04:48.076 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":29,"Reason":"StoreProhibited","EPC":"40094a1c","EXCVADDR":"0000003c","CallChain":["40094a19","40094e77","40094f78","4008527a","400852cc","4009534d","400e10f4","400e6bed","40161a82","401359ab","4010d1db","4015c56c","401675d2","400e7142"]},"BootCount":12}}
14:04:48.258 HTP: Main Menu
14:04:48.619 WIF: IPv6 Local fe80::3ee9:eff:feae:1a64
14:04:50.887 QPC: Reset
14:04:52.908 APP: Boot Count 12
14:04:52.913 RSL: STATE = {"Time":"2023-08-30T14:04:52","Uptime":"0T00:00:09","UptimeSec":9,"Heap":128,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"Simon","BSSId":"98:DA:C4:A4:46:F2","Channel":6,"Mode":"11n","RSSI":100,"Signal":-50,"LinkCount":1,"Downtime":"0T00:00:03"}}
14:04:53.678 CFG: Saved, Count 75, Bytes 4096
14:04:57.884 RSL: config = {"ip":"192.168.10.128","dn":"Tasmota","fn":["Tasmota",null,null,null,null,null,null,null],"hn":"tasmota-AE1A64-6756","mac":"3CE90EAE1A64","md":"ESP32-DevKit","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"13.1.0.1","t":"tasmota_AE1A64","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"bat":0,"dslp":0,"sho":[],"sht":[],"ver":1}
14:04:57.887 RSL: sensors = {"sn":{"Time":"2023-08-30T14:04:57"},"ver":1}
14:05:07.018 WIF: Checking connection...
```

Is there anything here, pointing to an issue preventing Tasmota to accept this template? 

I then tried to modify the template and put all RED lines to NONE and removed the RFRecv.  
It does not work either.

**Reply by sfromis** [reply]
Bad news:

> FRC: Some settings have been reset (6)

Tasmota has been boot looping, and disabled features until able to boot without crashing. Watching log via serial is a good way to see what happens.

**Reply by SimonFili** [reply]
Ha!  I see it now!  I'm trying to connect the ESP32 with serial, but no luck so far, need to research this a bit more.  Any good resources for that.  What I have now is a UART to serial, trying to use Putty to connect, but I have access denied.  Bummer! 

SCRAP THAT, I'm connected!  I forgot to check the baudrate command on the console!

**Reply by SimonFili** [reply]
Ok, here's the output of the serial log 4, and I see the "Core 1 panic'ed" line.  Does that help in any way understanding why this template make the core panic? 

```
14:57:00.026 CMD: module 0
14:57:00.028 SRC: WebConsole from 192.168.10.110
14:57:00.031 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 1, Pld 0, Data '0'
14:57:00.035 RSL: RESULT = {"Module":{"0":"ESP32R4"}}
14:57:01.102 CFG: Saved, Count 88, Bytes 4096
14:57:01.971 APP: Restarting
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DOUT, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12352
load:0x40080400,len:2912
entry 0x400805c4

00:00:00.003 HDW: ESP32-D0WD (PSRAM disabled)
00:00:00.004 HDW: PSRAM is disabled, requires specific compilation on this hardware (see doc)
00:00:00.051 UFS: FlashFS mounted with 12572 kB free
00:00:00.072 CFG: Loaded from File, Count 88
00:00:00.082 QPC: Count 1
00:00:00.084 CFG: CR 380/699, Busy 0
00:00:00.094 I2C: Bus1 using GPIO22(SCL) and GPIO21(SDA)
00:00:00.098 I2C: PCF8574 found at 0x21
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x4009494a  PS      : 0x00060033  A0      : 0x80094e7a  A1      : 0x3ffd1d50
A2      : 0x65d0d616  A3      : 0x00000000  A4      : 0x3ffb9a34  A5      : 0x00000000
A6      : 0x00060023  A7      : 0x00060020  A8      : 0x00000002  A9      : 0x0000000d
A10     : 0x0000003c  A11     : 0x20440fca  A12     : 0x3ffb9a58  A13     : 0x3ffb9a80
A14     : 0x00060023  A15     : 0x00060023  SAR     : 0x0000001f  EXCCAUSE: 0x0000001d
EXCVADDR: 0x0000000c  LBEG    : 0x4008d275  LEND    : 0x4008d289  LCOUNT  : 0xfffffff4


Backtrace: 0x40094947:0x3ffd1d50 0x40094e77:0x3ffd1d70 0x40094f78:0x3ffd1d90 0x4008527a:0x3ffd1db0 0x400852cc:0x3ffd1dd0 0x4009534d:0x3ffd1df0 0x400e6f6e:0x3ff                                                      d1e10 0x40161a82:0x3ffd1e80 0x401359ab:0x3ffd1ee0 0x4010d1db:0x3ffd1f30 0x4015c56c:0x3ffd1f50 0x401675d2:0x3ffd1f70 0x400e7142:0x3ffd2070




ELF file SHA256: 0000000000000000

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DOUT, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12352
load:0x40080400,len:2912
entry 0x400805c4

00:00:00.003 HDW: ESP32-D0WD (PSRAM disabled)
00:00:00.004 HDW: PSRAM is disabled, requires specific compilation on this hardware (see doc)
00:00:00.051 UFS: FlashFS mounted with 12572 kB free
00:00:00.072 CFG: Loaded from File, Count 88
00:00:00.074 FRC: Some settings have been reset (2)
00:00:00.075 CFG: CR 380/699, Busy 0
00:00:00.086 I2C: Bus1 using GPIO22(SCL) and GPIO21(SDA)
00:00:00.089 I2C: PCF8574 found at 0x21
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x4009494a  PS      : 0x00060033  A0      : 0x80094e7a  A1      : 0x3ffd1d50
A2      : 0x65d0d616  A3      : 0x00000000  A4      : 0x3ffb9a34  A5      : 0x00000000
A6      : 0x00060023  A7      : 0x00060020  A8      : 0x00000002  A9      : 0x0000000d
A10     : 0x0000003c  A11     : 0x20440fca  A12     : 0x3ffb9a58  A13     : 0x3ffb9a80
A14     : 0x00060023  A15     : 0x00060023  SAR     : 0x0000001f  EXCCAUSE: 0x0000001d
EXCVADDR: 0x0000000c  LBEG    : 0x4008d275  LEND    : 0x4008d289  LCOUNT  : 0xfffffff4


Backtrace: 0x40094947:0x3ffd1d50 0x40094e77:0x3ffd1d70 0x40094f78:0x3ffd1d90 0x4008527a:0x3ffd1db0 0x400852cc:0x3ffd1dd0 0x4009534d:0x3ffd1df0 0x400e6f6e:0x3ff                                                      d1e10 0x40161a82:0x3ffd1e80 0x401359ab:0x3ffd1ee0 0x4010d1db:0x3ffd1f30 0x4015c56c:0x3ffd1f50 0x401675d2:0x3ffd1f70 0x400e7142:0x3ffd2070




ELF file SHA256: 0000000000000000

```

**Comment by arendst** [comment]
OK. I can reproduce:
```
00:00:00.002-227/53 HDW: ESP32-D0WDQ6 
00:00:00.034-226/53 UFS: FlashFS mounted with 308 kB free
00:00:00.058-226/53 CFG: Loaded from File, Count 4298
00:00:00.060-225/53 FRC: Some settings have been reset (4)
00:00:00.066-224/53 I2C: Bus1 using GPIO21(SCL) and GPIO05(SDA)
00:00:00.068-224/53 I2C: PCF8574 found at 0x20
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400942c4  PS      : 0x00060433  A0      : 0x80094722  A1      : 0x3ffb1f30  
A2      : 0x3ffb2f90  A3      : 0xa5a5a5a5  A4      : 0x3ffb0014  A5      : 0x00000004  
A6      : 0x00060423  A7      : 0x00060420  A8      : 0x0000000d  A9      : 0x0000000b  
A10     : 0x3ffb03cc  A11     : 0x3ffb2fd0  A12     : 0x3ffb0038  A13     : 0x3ffb0060  
A14     : 0x3ffb2f98  A15     : 0x0000000e  SAR     : 0x0000000e  EXCCAUSE: 0x0000001d  
EXCVADDR: 0xa5a5a5b1  LBEG    : 0x4008d8e9  LEND    : 0x4008d8f9  LCOUNT  : 0xfffffff4  

```
Let me chk where it goes wrong....

**Comment by arendst** [comment]
Diving in the code I think Mode1 could never co-exist with GPIO relays; it tries to index an array to short dimensionend based on available relays.

I suggest you use Mode2 by keeping below commented:
```
//#define USE_PCF8574_SENSOR
//#define USE_PCF8574_MQTTINPUT
```
but enable:
```
#define USE_PCF8574_MODE2
```
once restarted add this as rule 3:
```
rule3 on file#pcf8574.dat do {"NAME":"PCF8574 A=Ri5-12","BASE":1,"GPIO":[260,261,262,263,264,265,266,267]} endon
```
and then perform a restart (command ``restart 1``

**Reply by SimonFili** [reply, marked as answer]
@arendst, you're the MEN!  (Let me know if I can buy you coffee!) 

I had to adjust the rule3 to this:
rule3 on file#pcf8574.dat do {"NAME":"PCF8574 A=R5-12","GPIO":[224,225,226,227,228,229,230,231]} endon

This is what works with the 4 relay on GPIO and the 8 extender relay on the i2c PCF8574.

So to recap:
1) I compiled on gitpod

2) modified /tasmota/my_user_config.h
2.1) removed comments here:
  #define USE_PCF8574
    #define USE_PCF8574_MODE2
2.2) added the EZO chip I need:
#define USE_EZOPH
#define USE_EZOORP
#define USE_EZOEC 
#define USE_EZODO
2.3) Removed some other module I don't need (for fun)
2.4) in /platform_overide_sample.ini
commented tasmota
uncommented tasmota32

3) Compiled with : platformio run -e tasmota32

4) took the firmware in /build_output/firmware/tasmota32.bin
 - uploaded to the ESP32
 
5)  Added through the command line, to configure the relay on the [PCF8574 extender board in mode2](https://tasmota.github.io/docs/PCF8574/#mode-2)
rule3 on file#pcf8574.dat do {"NAME":"PCF8574 A=R5-12","GPIO":[224,225,226,227,228,229,230,231]} endon
restart 1

6) this is the template I used (in other paramters)
{"NAME":"ESP32R4","GPIO":[1,1,576,1,480,1,1,1,1,1,1,1,1,1,1,1,0,640,608,1,0,224,225,0,0,0,0,0,227,226,32,33,34,0,0,35],"FLAG":0,"BASE":1}

Hope this help other use this kit from Rocket Controller

**Reply by sfromis** [reply]
One detail: In general, it is not really recommended to modify `my_user_config.h`, but instead make amendments in `user_config_override.h` - this way it is simpler to use same changes for a new build later. And when you specify build environment on `platformio run` (the recommended way), you do not need to modify `platform_override_sample.ini` to also select default build environment there.

**Reply by SimonFili** [reply]
Thanks, still learning!

**Comment by arendst** [comment]
With version v13.1.0.2 mode 1 is fixed solving the initial issue from OP too.

**Comment by Bettman66** [comment]
I think the problem is not solved with 13.1.0.2. I can't see the PCF8574 menu in Mode1.
It almost looks like he's in mode2, because it is looking for the file 'mcp23x.dat'.
The config worked with an older version.
Config:
```
#ifndef USE_I2C
#define USE_I2C
#define I2CDRIVERS_0_31        0xFFFFFFFF          // Enable I2CDriver0  to I2CDriver31
#define I2CDRIVERS_32_63       0xFFFFFFFF          // Enable I2CDriver32 to I2CDriver63
#define I2CDRIVERS_64_95       0xFFFFFFFF          // Enable I2CDriver64 to I2CDriver95
  #define USE_PCF8574                            // [I2cDriver2] Enable PCF8574 I/O Expander (I2C addresses 0x20 - 0x26 and 0x39 - 0x3F) (+1k9 code)
    #define USE_PCF8574_SENSOR                   // enable PCF8574 inputs and outputs in SENSOR message
    #define USE_PCF8574_DISPLAYINPUT             // enable PCF8574 inputs display in Web page
    #define USE_PCF8574_MQTTINPUT  
#endif
```
Log:
```
00:00:00.003 HDW: ESP32-D0WDQ6-V3 
00:00:00.043 UFS: FlashFS mounted with 304 kB free
00:00:00.068 CFG: Loaded from File, Anzahl 2331
00:00:00.078 QPC: Count 1
00:00:00.084 I2C: Bus1 using GPIO32(SCL) and GPIO33(SDA)
00:00:00.090 TFS: File 'mcp23x.dat' not found
00:00:00.171 BRY: Berry initialized, RAM used=4438 bytes
00:00:00.187 TFS: File '.drvset003' not found
00:00:00.195 Projekt tasmota - Nathalie.Relais Version 13.1.0.2(tasmota32)-2_0_11(2023.09.05 19:24:46)
00:00:00.979 WIF: verbinden mit AP1 1GBit Channel 6 BSSId xxxxxxx in Modus 11n als Nathalie.Relais...
00:00:02.951 WIF: verbunden
00:00:03.163 HTP: Web-Server aktiv bei Nathalie.Relais mit IP-Adresse xxxxxxxx
19:34:38.155 MQT: Verbindungsversuch...
19:34:38.202 MQT: verbunden
19:34:38.207 MQT: tele/Nathalie.Relais/LWT = Online (beibehalten)
19:34:38.208 MQT: cmnd/Nathalie.Relais/POWER = 
19:34:38.214 MQT: tele/Nathalie.Relais/INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"13.1.0.2(tasmota32)","FallbackTopic":"cmnd/Nathalie.Relais_fb/","GroupTopic":"cmnd/tasmotas/"}}
19:34:38.222 MQT: tele/Nathalie.Relais/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":40}}
19:34:40.954 QPC: Reset
19:34:42.957 MQT: tele/Nathalie.Relais/STATE = {"Time":"2023-09-05T19:34:42","Uptime":"0T00:00:09","UptimeSec":9,"Heap":157,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":1,"Berry":{"HeapUsed":4,"Objects":46},"Wifi":{"AP":1,"SSId":"xxx","BSSId":"xxxxxx","Channel":6,"Mode":"11n","RSSI":96,"Signal":-52,"LinkCount":1,"Downtime":"0T00:00:04"}}
19:34:59.448 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:34:59.499 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:34:59.547 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:34:59.600 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:34:59.647 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:34:59.698 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:35:20.210 CMD: i2cscan
19:35:20.236 MQT: stat/Nathalie.Relais/I2CSCAN = {"I2CScan":"Device(s) found on bus1 at 0x20"}
19:35:59.642 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:35:59.690 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:35:59.739 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:35:59.789 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:35:59.841 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
19:35:59.892 MQT: stat/Nathalie.Relais/POWER = {"Command":"Error"}
```

**Comment by Bettman66** [comment]
sorry my config was wrong, it works.
without #ifndef USE_I2C it worked

```
  #define USE_PCF8574                            // [I2cDriver2] Enable PCF8574 I/O Expander (I2C addresses 0x20 - 0x26 and 0x39 - 0x3F) (+1k9 code)
    #define USE_PCF8574_SENSOR                   // enable PCF8574 inputs and outputs in SENSOR message
    #define USE_PCF8574_DISPLAYINPUT             // enable PCF8574 inputs display in Web page
    #define USE_PCF8574_MQTTINPUT
```

**Comment by hmplima** [comment]
In my case the same thing happens. Simply program the esp32 with GPIO34/35/36/39 with a button, after a short time it restarts and is left with the default configuration

**Reply by barbudor** [reply]
If it restart and resets after 40 seconds, it is not a bug but a feature
RTFM: https://tasmota.github.io/docs/Device-Recovery/

**Reply by hmplima** [reply]
But the GPIO34/35/36/39 on the esp32 can be configured as input, but after programming them the esp32 restarts and remains with the default programming.
In this case, can I not program these GPIO as input?

**Reply by sfromis** [reply]
What exactly do you mean by "input"? Tasmota config as a "Button"?

**Reply by hmplima** [reply]
Yes, when I configure it as button

**Reply by sfromis** [reply]
What barbudor alluded to was that a pin configured as a Button, and registered as "pushed" for 40 seconds, will activate a recovery feature resetting to initial state. You should not configure pins without making sure that the electrical connection works for what you want. Maybe your inputs should be configured as "Switch" instead of "Button", or maybe as inverted Button, depending on how you want to use them.

**Reply by hmplima** [reply]
ok, I'll try it this way. Thank you for your help

**Comment by phildabill** [comment]
Just popping here to say I've had the exact same issue with my new ESP32R4. The device fully reset itself and lost all configuration (even the relays). I was only using relay 1 via an automation using MQTT. 

What information do you need to confirm if this is the same problem? The device is running:
Program Version	13.3.0(tasmota32)
Build Date & Time	2024-07-07T09:54:12
Core/SDK Version	2_0_14/4.4.6.231122

Oddly enough, I've had that exact same issue (device fully resetting) with cheaper 1 relay ESP8266 controllers running latest tasmota. It was the reason I purchased the ESP32R4 initially.

**Reply by sfromis** [reply]
Well, without any details of your config or operation, this does not mean much. Tasmota has recovery features *designed* to let you trigger a full reset, and you may have unintentionally activated one of them. https://tasmota.github.io/docs/Device-Recovery/
A typical example is a gpio configured as a button, and pulled down for 40 seconds.

**Reply by phildabill** [reply]
If this can help:

The cheaper board that had the same issue is that one: https://templates.blakadder.com/ESP12F_Relay_X1_V2.html

{"NAME":"LC-Relay-ESP12-1R-D8","GPIO":[255,255,157,255,255,21,255,255,255,255,255,255,57],"FLAG":15,"BASE":18}

The relay port is on IO5 if that plays a role (noticed arendts post with code showing I2C using GPIO05).

Only thing that I configure on the device is rename it and connect to to my MQTT broker to turn the relay on/off. No other modification is brought to it.

I reflashed it using Program Version 14.3.0 (release-tasmota-4M). We'll see if the same issue occurs within the next few days.

I will try to do the same with the ESP32R4.
